VisualRadio/app/src/main/java/altus/visualradio/MainListingActivity.java

```diff
@@ -0,0 +1,142 @@
+    //Read file into JSON array
+    public void readMsgFeedFile(String index_filename) throws IOException, JSONException {
+        //Assign file with name and directory of the VS_index_feed.json, which was created manually earlier
+        File msg_feed_file = new File(getExternalFilesDir(null), index_filename);
+        json_object = new JSONObject();
+        JSONArray temp_array = new JSONArray();
+        String temp_string = "";
+
+        if (msg_feed_file != null) {
+            //Create new file input stream
+            FileInputStream file_input_stream = new FileInputStream(msg_feed_file);
+            BufferedReader reader = new BufferedReader(new InputStreamReader(file_input_stream));
+            StringBuilder string_builder = new StringBuilder();
+            String current_line = null;
+            //Read every line till end of file
+            while ((current_line = reader.readLine()) != null) {
+                string_builder.append(current_line);
+            }
+            reader.close();
+            file_input_stream.close();
+            temp_string = string_builder.toString();
+            //Parse string from file through JSON tokener to extract characters and tokens properly
+            JSONTokener jsonTokener = new JSONTokener(temp_string);
+            //feed the parsed string into JSONArray
+            temp_array = new JSONArray(jsonTokener);
+        }
+        json_object.put(JSON_INDEX_KEY, temp_array);
+    }
+
+    //TODO
+    public void writeToIndexList() throws JSONException {
+        index_detail_setter = new ArrayList<>();
+        JSONArray temp_assignList_array = json_object.getJSONArray(JSON_INDEX_KEY);
+
+        for(int i = 0; i<temp_assignList_array.length(); i++){
+            ListDetailSetter listDetailSetter = new ListDetailSetter();
+            listDetailSetter.setTitle(temp_assignList_array.getJSONObject(i).getString("title"));
+
+            index_detail_setter.add(listDetailSetter);
+        }
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        // Inflate the menu; this adds items to the action bar if it is present.
+        getMenuInflater().inflate(R.menu.menu_main_listing, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        // Handle action bar item clicks here. The action bar will
+        // automatically handle clicks on the Home/Up button, so long
+        // as you specify a parent activity in AndroidManifest.xml.
+        int id = item.getItemId();
+
+        //noinspection SimplifiableIfStatement
+        if (id == R.id.action_settings) {
+            return true;
+        }
+
+        return super.onOptionsItemSelected(item);
+    }
+
+    protected void onListItemClick(ListView index_list, View v, int list_position, long id){
+        //Start intent that will make use of the InflatedViewActivity class
+        Intent inflate_view = new Intent(this, InflatedViewActivity.class);
```

Comment:
That was auto created, upon the creation of the activity. Since I have not changed anything to menu functionality, I left it as is.